### DISCUSSION TITLE
ClipToBounds=False and BoxShadow performance considerations?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#11386] [Q&A] [answered]
Hi, 

I have a performance questions for Avalonia Team.

ClipToBounds is True by default on most of the controls, I suppose it is for performance reason.

What would be the most efficient way (and not too ugly) to have a 1px shadow below a button?

1) if I set ClipToBounds=False but Clip to a rectangle geometry with button Bounds with 1 more pixel below, is it more efficient than ClipToBounds=False and no Clip at all? Is it lot worse than ClipToBounds=True? Is there a alternate better way?

2) is 1 pixels shadow such as "0 1 0 #color" slower than for example a border with border thickness of 0,0,0,1? I know that shadows are very slow in WPF, but I am not sure in Avalonia, may be they are cached internally and so performance free?

When I say ugly, I would like to avoid have the shadows in the button Bounds, for layout reason.

Could the Team give me guidance about that? Thanks a lot.

**Comment by kekekeks** [comment, marked as answer]
I'm not sure if there is still a performance reason for clipping everything. For buttons you can just disable ClipToBounds entirely.

**Reply by dbriard** [reply]
Thank you, good to know that the engine take care of that!
